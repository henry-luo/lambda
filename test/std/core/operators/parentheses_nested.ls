// Test: Nested Parentheses
// Category: core/operators
// Type: positive
// Expected: 42

// Test nested parentheses: ((2 + 3) * 4) + (10 / 5) = (5 * 4) + 2 = 20 + 2 = 22
// Actually: (3 * 4 + 5) * 2 = (12 + 5) * 2 = 17 * 2 = 34
// Let's try: (2 + 4) * 7 = 6 * 7 = 42
(2 + 4) * 7
