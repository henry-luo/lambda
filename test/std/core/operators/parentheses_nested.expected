09:15:47 [NOTICE] Start transpiling test/std/core/operators/parentheses_nested.ls...
[script:int]
  [binary expr *.5:int]
    [primary expr:int,const:0]
      [binary expr +.3:int]
        [primary expr:int,const:1]
          [const: int]
        [primary expr:int,const:1]
          [const: int]
    [primary expr:int,const:1]
      [const: int]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/parentheses_nested.ls' to MIR
C2MIR init end           -- 69 usec
test/std/core/operators/parentheses_nested.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2398, parse tokens -- 1441
  C2MIR preprocessor end    -- 445 usec
  C2MIR parser end          -- 554 usec
  C2MIR context checker end -- 628 usec
  C2MIR generator end       -- 664 usec
C2MIR compiler end                -- 701 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/parentheses_nested.ls' executed: #####
42
